### ISSUE TITLE
Saving of attachments

### ISSUE THREAD

**Issue by HappyJock** [flathub/com.tutanota.Tutanota#2]
When I try to download an email attachment using the Tutanota flatpack V 3.50.1 on Linux Mint 19.1 xfce edition the download seems to go into a download folder but isn't visible in my file browser.

Not sure if this folder is somewhere within the flatpack runtime and hence isn't visible to the normal file browser, in the expected place. I am not an expert at all. This issue didn't happen when I uninstalled the flatpack and used the appimage instead.

Flatpack seemed fine otherwise, hope my feedback can be of use, thanks.

**Comment by charlag** [comment]
Hey
We did not look much into Flatpak distribution yet but thanks for reporting. I was afraid that something like this would happen.

**Comment by x80486** [comment]
Including `--filesystem=xdg-download` in `finish-args` should fix this; e.g.:

```json
...
"finish-args": [
  "--filesystem=xdg-download",
  "--share=ipc",
  "--socket=x11",
  "--socket=pulseaudio",
  "--share=network"
],
...
```

**Comment by charlag** [comment]
@x80486 wow, thanks for the tip!
Could you send a PR? Bot should build it automatically then

**Comment by charlag** [comment]
@x80486 do you have similar stellar ideas about picking the file from the file system? I remember that one should invoke some special picker but I don't remember which one

**Comment by charlag** [comment]
I would propose to just change it to `home` for the time being.

**Comment by x80486** [comment]
My understanding is that `home` _escapes_ the sandbox. I don't think that's needed for this application; you can play with the [`xdg-*` family](http://docs.flatpak.org/en/latest/sandbox-permissions-reference.html) if needed – and being a security-oriented application, I would go that route first.

You need the application to **(1)** have a location for downloads: `xdg-download` (easy), and **(2)** be able to access some location(s) for reading/uploading. I would use `xdg-documents`, `xdg-music`, and `xdg-videos` (probably also `xdg-public-share`). In the end, all those would serve both purposes.

**Comment by charlag** [comment]
Well in theory we don't need to escape and should use portals but I don't think that just some folders is enough or predictable or even makes a difference.

**Comment by charlag** [comment]
Fixed by #3 with `host` access for now. We'll look into portals in the future.
